flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/AsyncSinkWriter.java

```diff
@@ -256,14 +243,42 @@ public AsyncSinkWriter(
             long maxTimeInBufferMS,
             long maxRecordSizeInBytes,
             Collection<BufferedRequestState<RequestEntryT>> states) {
+        this(
+                elementConverter,
+                context,
+                maxBatchSize,
+                maxBufferedRequests,
+                maxBatchSizeInBytes,
+                maxTimeInBufferMS,
+                maxRecordSizeInBytes,
+                states,
+                CongestionControlRateLimitingStrategy.builder()
+                        .setMaxInFlightRequests(maxInFlightRequests)
+                        .setInitialMaxInFlightMessages(maxBatchSize)
+                        .setAimdScalingStrategy(
+                                AIMDScalingStrategy.builder()
+                                        .setRateThreshold(maxBatchSize * maxInFlightRequests)
+                                        .build())
+                        .build());
+    }
+
+    public AsyncSinkWriter(
+            ElementConverter<InputT, RequestEntryT> elementConverter,
+            Sink.InitContext context,
+            int maxBatchSize,
+            int maxBufferedRequests,
+            long maxBatchSizeInBytes,
+            long maxTimeInBufferMS,
+            long maxRecordSizeInBytes,
+            Collection<BufferedRequestState<RequestEntryT>> states,
+            RateLimitingStrategy rateLimitingStrategy) {
```

Comment:
Does the `AsyncSinkWriter` have any documentation? How users are supposed to know that it exists and especially what kind of `RateLimitingStrategy` they can use?

Reply:
We're writing a blog post about it instead (for now) https://issues.apache.org/jira/browse/FLINK-25208